test/client/components/CommunityPosts.test.js

```diff
@@ -30,82 +34,91 @@ const currentUser = {
 }
 
 describe('CommunityPosts', () => {
-  let store
-
   beforeEach(() => {
-    store = configureStore({
-      communities: {[community.slug]: community},
-      people: {
-        current: currentUser
-      }
-    }).store
     window.FEATURE_FLAGS = {
       REQUEST_TO_JOIN_COMMUNITY: 'on',
       COMMUNITY_SETUP_CHECKLIST: 'on',
       IN_FEED_PROFILE_COMPLETION_MODULES: 'on'
     }
   })
 
-  it('renders without errors', () => {
+  const setupNode = (setupCommunity = community, setupCurrentUser = currentUser) => {
+    const store = configureStore({
+      communities: {[setupCommunity.slug]: setupCommunity}
+    }).store
     const props = {
-      params: {id: community.slug},
+      params: {id: setupCommunity.slug},
       location: {}
     }
-
-    const node = mount(<CommunityPosts {...props} />, {
-      context: {store, currentUser},
+    return mount(<CommunityPosts {...props} />, {
+      context: {store, currentUser: setupCurrentUser},
       childContextTypes: {currentUser: PropTypes.object}
     })
-    mockActionResponse(updateCurrentUser(), {})
+  }
 
+  it('renders without errors', () => {
+    const node = setupNode(community)
     expect(node.find('.community-setup').length).to.equal(1)
-
     expect(node.find('.post-editor').length).to.equal(1)
-
     expect(node.find('.no-results').length).to.equal(1)
   })
 
   it('displays request to join and no PostEditor, when not a member', () => {
-    const props = {
-      params: {id: community.slug},
-      location: {}
-    }
-
-    const node = mount(<CommunityPosts {...props} />, {
-      context: {store, currentUser: {id: 6}},
-      childContextTypes: {currentUser: PropTypes.object}
-    })
-
+    const nonCommunityMember = {id: 6}
+    const node = setupNode(community, nonCommunityMember)
     expect(node.find('.post-editor').length).to.equal(0)
-
     expect(node.find('.request-to-join').text())
     .to.equal('You are not a member of this community. Request to Join')
   })
 
+  describe('EngagementModules', () => {
```

Comment:
if it doesn't refer to an actual object let's use normal words instead of camelcase